proto/pulumi/provider.proto

```diff
@@ -530,6 +530,10 @@ message CheckRequest {
         Mode mode = 2;
     }
     AutonamingOptions autonaming = 8;
+
+    // The organization to which this resource will belong. The project and stack can be retrieved through
+    // the resource's URN, but the organization is not part of this and cannot, so it requires its own field.
+    string organization = 9;
```

Comment:
We should possibly add Project and Stack directly as well. It saves us having to implement URN parse logic in each SDK language. I added Name and Type a while ago for that reason (see fields 6 and 7).

Reply:
In hindsight a URN message struct with the full value and then each parsed out value sent to every resource method would probably have been best.
```
message Urn {
   string urn = 1;
   string name = 2;
   string type = 3;
...
} 

message CheckRequest {
   Urn urn = 1;
   ...
}

message DeleteRequest {
   Urn urn = 1;
   ...
}

etc
```

Not sure that's worth doing now rather than just continuing to inline all the fields, given half of them are already inlined, but maybe worth a thought.